lib/Conversion/TorchToLinalg/IndirectDataMovement.cpp

```diff
@@ -168,6 +168,270 @@ class ConvertAtenEmbeddingOp : public OpConversionPattern<AtenEmbeddingOp> {
+      return rewriter.notifyMatchFailure(
+          op, "sparse is expected to be a constant boolean value.");
+    }
+
+    if (isSparse) {
+      return op.emitError(
+          "Unimplemented: Sparse mode is not supported yet for EmbeddingBag.");
+    }
+
+    bool discard_last_offset;
+    if (!matchPattern(include_last_offset,
+                      m_TorchConstantBool(&discard_last_offset))) {
+      return rewriter.notifyMatchFailure(
+          op,
+          "include_last_offset is expected to be a constant boolean value.");
+    }
+
+    auto weightTy = weight.getType().cast<RankedTensorType>();
+    if (weightTy.getRank() != 2)
+      return rewriter.notifyMatchFailure(op, "weight must be rank 2");
+
+    auto indicesTy = indices.getType().cast<RankedTensorType>();
+    if (indicesTy.getRank() != 1)
+      return rewriter.notifyMatchFailure(op, "indices must be a vector");
+
+    auto offsetsTy = offsets.getType().cast<RankedTensorType>();
+    if (offsetsTy.getRank() != 1)
+      return rewriter.notifyMatchFailure(op, "offsets much be a vector");
+
+    Type weightElemTy = weightTy.getElementType();
+
+    if (modeInt == torch_upstream::EmbeddingBagMode::MODE_SUM) {
+      int64_t iterationMapDimension = weightTy.getRank() + indicesTy.getRank();
+
+      SmallVector<AffineExpr> indicesExpr;
+      indicesExpr.push_back(mlir::getAffineDimExpr(1, context));
+
+      auto indicesIndexingMap =
+          AffineMap::get(/*dimCount=*/iterationMapDimension, /*symbolCount=*/0,
+                         indicesExpr, context);
+
+      SmallVector<AffineExpr> offsetsExpr;
+      offsetsExpr.push_back(mlir::getAffineDimExpr(0, context));
+
+      auto offsetIndexingMap =
+          AffineMap::get(/*dimCount=*/iterationMapDimension, /*symbolCount=*/0,
+                         offsetsExpr, context);
+
+      SmallVector<AffineExpr> outputExpr;
+      outputExpr.push_back(mlir::getAffineDimExpr(0, context));
+      outputExpr.push_back(mlir::getAffineDimExpr(2, context));
+
+      auto outputIndexingMap =
+          AffineMap::get(/*dimCount=*/iterationMapDimension, /*symbolCount=*/0,
+                         outputExpr, context);
+
+      SmallVector<AffineMap, 3> indexingMaps = {
+          indicesIndexingMap,
+          offsetIndexingMap,
+          outputIndexingMap,
+      };
+
+      SmallVector<StringRef> iteratorTypes(iterationMapDimension,
+                                           getParallelIteratorTypeName());
+
+      Value embeddingDim = getDimOp(rewriter, loc, weight, 1);
+
+      Value initTensor;
+      Value offsetsLength;
+      Value indicesLength;
+      if (!discard_last_offset) {
+
+        SmallVector<Value> sizes{getDimOp(rewriter, loc, offsets, 0),
+                                 embeddingDim};
+
+        initTensor = createZeroInitTensor(rewriter, loc, sizes, weightElemTy);
+
+        offsetsLength = getDimOp(rewriter, loc, offsets, 0);
+
+        indicesLength = getDimOp(rewriter, loc, indices, 0);
```

Comment:
nit: here we try to reduce the amount of whitespace that does not add to readability. In this case, adding empty lines between lines does not make much of a difference since every chunk of code is just one statement, so I would just remove all empty lines (including the one right after the `if`)